svgbasic.go

```diff
@@ -189,11 +190,65 @@ func pathParse(pathStr string) (segs []SVGBasicSegmentType, err error) {
 	return
 }
 
+type EnhancedSegment struct {
+	Commands  []SVGBasicSegmentType
+	FillState fillState
+}
+
 // SVGBasicType aggregates the information needed to describe a multi-segment
 // basic vector image
 type SVGBasicType struct {
 	Wd, Ht   float64
-	Segments [][]SVGBasicSegmentType
+	Segments []EnhancedSegment
+}
+
+type fillState struct {
+	isFilled     bool
+	fillStrategy string
+	r            int
+	g            int
+	b            int
+}
+
+// Parse style tags on path e.g. style="stroke: rgb(0,0,0); stroke-width: 1; stroke-dasharray: none; stroke-linecap: butt; stroke-dashoffset: 0; stroke-linejoin: miter; stroke-miterlimit: 4; fill: rgb(0,0,0); fill-rule: nonzero; opacity: 1;"
+func getFillState(style string) (fs fillState, err error) {
+	rules := strings.Split(style, ";")
+	fs.isFilled = false
+	for _, rule := range rules {
+		parsedRule := strings.Split(rule, ":")
+		key := strings.TrimSpace(parsedRule[0])
+		if key == "fill" {
+			fs.isFilled = true
+			fillColor := strings.TrimSpace(parsedRule[1])
+			rgbr := regexp.MustCompile(`rgb\(*([0-9]+),*([0-9]+),*([0-9]+)\)`)
+			colors := rgbr.FindStringSubmatch(fillColor)
+			r, err := strconv.Atoi(colors[1])
+			if err != nil {
+				return fs, err
+			}
+			fs.r = r
+			g, err := strconv.Atoi(colors[2])
+			if err != nil {
+				return fs, err
+			}
+			fs.g = g
+			b, err := strconv.Atoi(colors[3])
+			if err != nil {
+				return fs, err
+			}
+			fs.b = b
+		}
+		if key == "fill-rule" {
+			fs.isFilled = true
+			fs.fillStrategy = strings.TrimSpace(parsedRule[1])
+		}
+	}
+	return fs, err
+}
+
+type pathType struct {
```

Comment:
same uppercase here, for what i could understood is the right pattern right @jsaguiar ?

Reply:
uppercase just means that somethign is exported. lowercase is private

Reply:
ohhh ok ok gotta get my golang game up xD